### ISSUE TITLE
 yara-python-dex

### ISSUE THREAD

**Issue by AhmadShakerASH** [MobSF/Mobile-Security-Framework-MobSF#1854]
ERROR: Could not find a version that satisfies the requirement yara-python-dex>=1.0.0 (from apkid==2.1.2->-r requirements.txt (line 24)) (from versions: none)
ERROR: No matching distribution found for yara-python-dex>=1.0.0 (from apkid==2.1.2->-r requirements.txt (line 24))

**Comment by kushal154** [comment]
@AhmadShakerASH Did you manage to resolve?

**Comment by idotene** [comment]
Same error here for a clean install

**Comment by uvicorn** [comment]
same

**Comment by Reeeick** [comment]
Install works when using python3.9 as per the installation requirements: https://mobsf.github.io/docs/#/requirements

Python3.10 breaks the dependency.

**Comment by DeadPackets** [comment]
You can install `yara-python-dex` directly from the repository by running the following command:

```
pip install -U git+https://github.com/MobSF/yara-python-dex
```

This solved the issue for me.

**Comment by rtsketo** [comment]
> I was facing the same issue while running `./setup.sh`. I was able to fix it by changing `apkid==2.1.2` to `apkid==2.1.1` in `requirements.txt` file.

This fixed it for me, but failed the setup later on. My declared version was `2.1.4`.
Installing with Python 3.9 is the only solution so far.

**Comment by vral-parmar** [comment]
i solve it using following and my mobsf is running flawlessly.

cd to mobsf and run ./setup.sh its give me following error 
<img width="635" alt="Screenshot 2024-07-03 at 11 34 45 AM" src="https://github.com/MobSF/Mobile-Security-Framework-MobSF/assets/31970912/19e9b74a-e231-4f45-81e1-dfe5e9babc50">
so i install poetry by following 
<img width="508" alt="Screenshot 2024-07-03 at 11 36 17 AM" src="https://github.com/MobSF/Mobile-Security-Framework-MobSF/assets/31970912/9a7a6624-a300-4d10-954f-9ca41d10efdc">
after installing run `poetry shell ` it will enter to the mobsf virtual environment after that use `pip freeze | "<package name>"` to verify missing package
<img width="646" alt="Screenshot 2024-07-03 at 11 37 23 AM" src="https://github.com/MobSF/Mobile-Security-Framework-MobSF/assets/31970912/509c13f7-0356-44b6-ac06-b227c3a69da0">
now manually install package using wheel 
my environment is missing yara so i build it locally using 
```
pip install wheel
pip wheel --wheel-dir=yara-python-dex git+https://github.com/MobSF/yara-python-dex.git
pip install --no-index --find-links=yara-python-dex yara-python-dex
```
Verify installed package `pip freeze | grep "<PackageName>"`

thats it exit environment and run ./run.sh 
<img width="637" alt="Screenshot 2024-07-03 at 11 42 04 AM" src="https://github.com/MobSF/Mobile-Security-Framework-MobSF/assets/31970912/b1904e8e-5718-4758-b6e4-645f4745b4d2">

this solution work for me!!
